### PR TITLE
refactor: remove extern/const where possible

### PR DESCRIPTION
**Pull request by ObserverOfTime**
`extern` is redundant in function prototypes.
`const` is discarded in most if not all binding functions.

### CODE REVIEW

**Review by amaanq — APPROVED** [review]

File: cli/src/generate/render.rs [hunk 1]
```diff
@@ -263,7 +263,6 @@ impl Generator {
 
     fn add_pragmas(&mut self) {
         add_line!(self, "#if defined(__GNUC__) || defined(__clang__)");
-        add_line!(self, "#pragma GCC diagnostic push");
```
Comment (maxbrunsfeld) [inline cli/src/generate/render.rs:266]: why this change?

Comment (ObserverOfTime) [reply cli/src/generate/render.rs:266]: No point in pushing if it's not popped. It will apply till the end of the file anyway.

Comment (maxbrunsfeld) [reply cli/src/generate/render.rs:266]: oh ok, sounds good.
